mesos-rxjava-recordio/src/test/java/org/apache/mesos/rx/java/recordio/ByteArrays.java

```diff
@@ -29,33 +30,31 @@
     private ByteArrays() {}
 
     @NotNull
-    static List<byte[]> partitionIntoArraysOfSize(@NotNull final byte[] allBytes, final int partSize) {
-        final List<byte[]> newChunks = new ArrayList<>();
-        byte[] newChunk = null;
-        for (int i = 0; i < allBytes.length; i++) {
-            final byte b = allBytes[i];
-            final int partIndex = i % partSize;
-            if (partIndex == 0) {
-                if (newChunk != null) {
-                    newChunks.add(newChunk);
-                }
-                newChunk = new byte[partSize];
-            }
-            //noinspection ConstantConditions
-            newChunk[partIndex] = b;
-
-            if (i + 1 == allBytes.length) {
-                newChunks.add(newChunk);
-            }
+    static List<byte[]> partitionIntoArraysOfSize(@NotNull final byte[] allBytes, final int chunkSize) {
+        final List<byte[]> chunks = new ArrayList<>();
+        final int numFullChunks = allBytes.length / chunkSize;
+
+        int chunkStart = 0;
+        int chunkEnd = 0;
+
+        for (int i = 0; i < numFullChunks; i++) {
+            chunkEnd += chunkSize;
+            chunks.add(Arrays.copyOfRange(allBytes, chunkStart, chunkEnd));
+            chunkStart = chunkEnd;
         }
-        return newChunks;
+
+        if (chunkStart < allBytes.length) {
+            chunks.add(Arrays.copyOfRange(allBytes, chunkStart, allBytes.length));
+        }
+
+        return chunks;
     }
 
     @NotNull
     static byte[] concatAllChunks(@NotNull final List<byte[]> chunks) {
         return chunks
             .stream()
-            .reduce(ZERO_BYTES, Bytes::concat/*acc, val*/);
+            .reduce(ZERO_BYTES, Bytes::concat);
```

Comment:
I added the comment here because the direction of the copy is significant (I had an early bug where I had flipped the `acc` and `val` and was only ever getting `0`s out the other side :(